rosbag2_transport/src/rosbag2_transport/player.hpp

```diff
@@ -59,6 +61,15 @@ class Player
   void play_messages_until_queue_empty(const PlayOptions & options);
   void prepare_publishers();
 
+  /**
+   * Determine which QoS to offer for a topic.
+   * Overrides the QoS for a topic if one is specified in the play_options.
```

Comment:
Does it though? As far as I can see this function only returns a QOS, but doesn't do any further actions such as overriding anything.